src/y_text.rs

```diff
@@ -56,30 +60,55 @@ impl YText {
         }
     }
 
+    /// Returns an underlying shared string stored in this data type.
+    pub fn __str__(&self) -> String {
+        match &self.0 {
+            SharedType::Integrated(v) => v.to_string(),
+            SharedType::Prelim(v) => v.clone(),
+        }
+    }
+
+    pub fn __repr__(&self) -> String {
+        format!("YText({})", self.__str__())
+    }
+
     /// Returns length of an underlying string stored in this `YText` instance,
     /// understood as a number of UTF-8 encoded bytes.
-    #[getter]
-    pub fn length(&self) -> u32 {
+    pub fn __len__(&self) -> usize {
         match &self.0 {
-            SharedType::Integrated(v) => v.len(),
-            SharedType::Prelim(v) => v.len() as u32,
+            SharedType::Integrated(v) => v.len() as usize,
+            SharedType::Prelim(v) => v.len(),
         }
     }
 
-    /// Returns an underlying shared string stored in this data type.
-    pub fn to_string(&self, txn: &YTransaction) -> String {
+    pub fn __iter__(&self) -> YTextIterator {
+        // Unsafe iterator references possible because blocks holding pointer results aren't ever deallocated.
+        let iterator = match &self.0 {
+            SharedType::Integrated(text) => unsafe {
+                let text_str = text.to_string();
```

Comment:
I'm not sure if this is a good approach. If you need to interate over string produced by YText, you can always just call `str(ytext)`. Using iterator this way ignores the fact, that YText may contain non-string elements (formatting attributes and embedded blocks).

Reply:
Since there isn't a standard way of iterating over the rich text, I'd be happy to help implement that in Yrs. I agree that `str(ytext)` is a fine approach for iterating over the characters. I can take out iteration for this PR.

Reply:
As of this PR, rich text hasn't been added to `YText`. I propose that we keep the current form of iteration over the string representation and discuss a new implementation with the fix to #7

Reply:
In theory there's a [Text::diff](https://github.com/y-crdt/y-crdt/blob/main/yrs/src/types/text.rs#L544) method, which returns additive changes from a given snapshot.